entrypoint.sh

```diff
@@ -2,31 +2,23 @@
 
 set -e
 
-# Escape double quotes in the params
-PARAMS_ESCAPED=`echo $PARAMETERS | sed 's/"/\\\"/g'`
-
-# Optional payload
-WORKSPACE=${TOWER_WORKSPACE_ID:+'workspaceId='$TOWER_WORKSPACE_ID}
-COMPUTE=${TOWER_COMPUTE_ENV:+'"computeEnvId": "'$TOWER_COMPUTE_ENV'",'}
-REV=${REVISION:+'"revision": "'$REVISION'",'}
-
-PAYLOAD='
-{
-    "launch": {
-        "pipeline": "'${PIPELINE}'",
-        "workDir": "'${WORKDIR}'",
-        '${COMPUTE}${REV}'
-        "paramsText": "'$PARAMS_ESCAPED'",
-        "configProfiles": '$CONFIG_PROFILES',
-        "resume": false
-    }
-}'
+PAYLOAD=$(
+    jq -n \
+        --arg pipeline "$PIPELINE" \
+        --arg workDir "$WORKDIR" \
+        ${TOWER_COMPUTE_ENV:+--arg computeEnvId "$TOWER_COMPUTE_ENV"} \
+        ${REVISION:+--arg revision "$REVISION"} \
+        --arg paramsText "$PARAMETERS" \
+        --arg configProfiles "$CONFIG_PROFILES" \
+        --argjson resume false \
+        '{ launch: $ARGS.named }'
+)
 
 curl \
     --silent \
     --show-error \
     --fail \
```

Comment:
New `curl` option available as of version `7.76.0`. When I build this Docker container locally I currently get version `7.79.1` so should be fine to use.

https://daniel.haxx.se/blog/2021/02/11/curl-fail-with-body/

```suggestion
    --fail-with-body \
```